fix(CI): remove self-executing tests from megatest

saem opened the pull request:
## Summary

Marked `texitsignal` as not joinable to exclude it from the `megatest`
category; this should hopefully limit intermitent CI failures.

## Details

`texitsignal` attempts to execute itself with different parameters and
uses that to branch logic of what to check. This doesn't fit with the
rest of the megatest category.

Also, marked these tests as not `joinable` as they either self-execute
or kickoff compiler processes:

- `texception_message_null_byte`
- `tstaticlib`
- `tstackframes`
- `treadlines`
- `tvmops`

Lastly, disabled `tclose` for macOS testing within the same category, as
it's linux only already to save some CI cycles.

saem committed:
fix(CI): remove self-executing tests from megatest

Summary

Marked `texitsignal` as not joinable to exclude it from the `megatest`
category; this should hopefully limit intermitent CI failures.

Details

`texitsignal` attempts to execute itself with different parameters and
uses that to branch logic of what to check. This doesn't fit with the
rest of the megatest category.

Also disabled `tclose` for macOS testing within the same category, as
it's linux only already to save some CI cycles.

zerbina commented on tests/stdlib/os/osproc/tclose.nim:
```diff
@@ -1,5 +1,6 @@
 discard """
   exitcode: 0
+  disable: osx
```
```suggestion
  disabled: osx
```
This should fix the test failure.

saem replied:
I was scouring the logs trying to spot an error and couldn't find the spec parse error. 😅 

Thanks.

zerbina reviewed (approved):
Another test that self-executes and is joined is the `texception_message_null_byte.nim` test. Marking it as unjoinable would also fix the megatest for the C target failing when no NodeJS installation is present.

saem committed:
typo: `disable` -> `disabled`

Co-authored-by: zerbina <100542850+zerbina@users.noreply.github.com>

saem committed:
Mark tests as not joinable

Also, marked these tests as not `joinable` as they either self-execute
or kickoff compiler processes:

- `texception_message_null_byte`
- `tstaticlib`
- `tstackframes`
- `treadlines`
- `tvmops`

saem commented:
> Another test that self-executes and is joined is the `texception_message_null_byte.nim` test. Marking it as unjoinable would also fix the megatest for the C target failing when no NodeJS installation is present.

Ah, thanks for the heads up. I did a quick search for `getCurrentCompiler` inside the test directory and marked a bunch more as `joinable: false`, I'm going to include them and update the PR description.

saem commented:
/merge

saem commented:
/merge
